kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/DatasetTarget.java

```diff
@@ -99,12 +105,12 @@ public boolean handleExisting(WriteMode writeMode, long lastModForSource,
           LOG.info("Writing to existing dataset/view: " + view);
           break;
         case CHECKPOINT:
-          boolean ready = true; // dataset.isReady(); // TODO: add when CDK-451 is ready
           long lastModForTarget = -1;
           if (view instanceof LastModifiedAccessor) {
             lastModForTarget = ((LastModifiedAccessor) view).getLastModified();
           }
-          if (ready && lastModForTarget > lastModForSource) {
+
+          if (ready && (lastModForTarget > lastModForSource)) {
```

Comment:
I think that makes sense. Since we aren't using an atomic operation to complete MR writes, we should use the signal.

In that case, it sounds like a useful thing to add always signal in the merge output format, like you had earlier. But, if the dataset implementation isn't Signalable, I think we should only consider the modification time. That way we don't break checkpoints for things that don't have signals implemented, like HBase.